app/forms/lead_providers/report_schools_form.rb

```diff
@@ -17,13 +17,20 @@ class ReportSchoolsForm
     def save!
       ActiveRecord::Base.transaction do
         school_ids.each do |school_id|
-          partnership = Partnership.create!(
+          partnership = Partnership.find_or_initialize_by(
             school_id: school_id,
             cohort_id: cohort_id,
-            delivery_partner_id: delivery_partner_id,
             lead_provider_id: lead_provider_id,
           )
 
+          partnership.challenge_reason = partnership.challenged_at = nil
+          partnership.delivery_partner_id = delivery_partner_id
+          partnership.save!
+
```

Comment:
I think we also need to update the challenge deadline here, otherwise it will not be possible to challenge the re-reported partnership. It's currently handled in a `before_create`

Reply:
There's something very strange going on here. We have both a `report_schools_form` and a `confirm_schools_form`, only one of which deals with pending partnerships

Reply:
In any case, we're missing the `after_create :schedule_activation` here for the following edge case:

1. Partnership reported for CIP school
2. School challenges within 14 days, remains CIP
3. Partnership re-reported at least 14 days after the first one

In this case, the school would never be transitioned from CIP to FIP

Reply:
Good catch! :)